File: ND_Telemedicine/lib/Models/Record.dart
class Record {
  String name;
  String surname;
  String gender;
  String email;
  String allergies;
  String status;

  Record(
     this.name,
     this.surname,
     this.gender,
     this.email,
     this.allergies,
     this.status
  );

  // factory Record.fromJson(Map<String, dynamic> json) {
  //   return Record(
  //       name: json['name'],
  //       surname: json['surname'],
  //       gender: json['gender'],
  //       email: json['email'],
  //       allergies: json['password'],
  //       status: json['status'],
  //   );
  // }
}

File: BackEnd/src/main/java/com/example/backend/model/Record.java
package com.example.backend.model;

import javax.persistence.*;

@Entity
public class Record {

    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    private Integer id;

    @Column
    private String name;

    @Column
    private String surname;

    @Column
    private String gender;

    @Column
    private String email;

    @Column
    private String allergies;

    @Column
    private String status;

    //getters and setters below
    //id
    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    //name(first/given name)
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    //surname
    public String getSurname() {
        return surname;
    }

    public void setSurname(String surname) {
        this.surname = surname;
    }

    //gender
    public String getGender() {
        return gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }

    //email
    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    //allergies
    public String getAllergies() {
        return allergies;
    }

    public void setAllergies(String allergies) {
        this.allergies = allergies;
    }

    //status
    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

}
